jdbc/src/main/java/org/apache/iotdb/jdbc/IoTDBQueryResultSet.java

```diff
@@ -1251,15 +1294,14 @@ private String getValueByName(String columnName) throws SQLException {
     if (columnName.equals(TIMESTAMP_STR)) {
       return String.valueOf(record.getTimestamp());
     }
-    int tmp = columnInfoMap.get(columnName);
-    int i = 0;
-    for (Field field : record.getFields()) {
-      i++;
-      if (i == tmp - 1) {
-        return field.getDataType() == null ? null : field.getStringValue();
-      }
+    int index = columnInfoMap.get(columnName) - START_INDEX;
+    Field field = record.getFields().get(index);
+    if (field != null) {
+      return field.getStringValue();
```

Comment:
In the old code, if `field.getDataType` equals null, then `null` is returned instead of a String valued "null". I think there is difference between a null String and a String whose value is "null", although the printed result is the same "null".

Reply:
Fixed. Thank you.